job_definitions/update_jenkins_job_definitions.yml

```diff
@@ -0,0 +1,35 @@
+- job:
+    name: "update-jenkins-job-definitions"
+    display-name: "Update Jenkins Job Definitions"
+    project-type: pipeline
+    concurrent: false
+    disabled: false
+    description: |
+      <p>This job updates Jenkins jobs using the job definitions from https://github.com/alphagov/digitalmarketplace-jenkins</p>
+    scm:
+      - git:
+          url: git@github.com:alphagov/digitalmarketplace-jenkins.git
+          credentials-id: github_com_and_enterprise
+          branches:
+            - "main"
+    triggers:
+      - pollscm:
+          cron: "H/2 * * * *"
+    wrappers:
+      - ansicolor
+    dsl: |
+      node {
+            stage('Clone digitalmarketplace-jenkins') {
+              git url: 'git@github.com:alphagov/digitalmarketplace-jenkins.git', branch: 'main', credentialsId: 'github_com_and_enterprise', poll: true 
+            }
+            stage('Update jobs') {
+              withEnv([
+                "DM_CREDENTIALS_REPO=/var/lib/jenkins/digitalmarketplace-credentials"
+                ]) {
+                 sh('make venv && \
+                 source venv/bin/activate && \
+                 pip3 install --upgrade pip && \
```

Comment:
@bjgill I found if I just run `make jobs` without manually upgrading pip first, I get the following error on Jenkins:
```
+ make jenkins TAGS=jobs
[ -z $VIRTUAL_ENV ] && [ ! -d venv ] && python3 -m venv venv || true
/data/jenkins/jobs/update-jenkins-job-definitions/workspace/venv/bin/pip install -Ur requirements.txt
Collecting ansible==3.1.0 (from -r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/19/0c/03233fb2896c6e04c403c7950eec4e2e41bafc059320d95926a95cce2c69/ansible-3.1.0.tar.gz
Collecting ansible-base<2.11,>=2.10.5 (from ansible==3.1.0->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/ba/58/009ab5eccdac9f44c52cc50391d5c8b986bc6559e0d56f3cc558a779ca48/ansible-base-2.10.7.tar.gz
Collecting PyYAML (from ansible-base<2.11,>=2.10.5->ansible==3.1.0->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/7a/5b/bc0b5ab38247bba158504a410112b6c03f153c652734ece1849749e5f518/PyYAML-5.4.1-cp36-cp36m-manylinux1_x86_64.whl
Collecting cryptography (from ansible-base<2.11,>=2.10.5->ansible==3.1.0->-r requirements.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/fa/2d/2154d8cb773064570f48ec0b60258a4522490fcb115a6c7c9423482ca993/cryptography-3.4.6.tar.gz
    Complete output from command python setup.py egg_info:
            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:
            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-lokt0xzz/cryptography/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-lokt0xzz/cryptography/
Makefile:28: recipe for target 'requirements' failed
make: *** [requirements] Error 1
```

Reply:
Makes sense. If `make jobs` needs an updated version of pip to run (`"pip>=19.0"` I believe), should this be in the Makefile?